src/extension-runners/index.js

```diff
@@ -0,0 +1,67 @@
+/* @flow */
+
+import type {
+  IExtensionRunner,  // eslint-disable-line import/named
+} from './base';
+
+export type MultipleTargetsExtensionRunnerParams = {
+  runners: Array<IExtensionRunner>,
+};
+
+// Export everything exported by the firefox-desktop runner.
+export * from './firefox-desktop';
+
+export class MultipleTargetsExtensionRunner {
+  extensionRunners: Array<IExtensionRunner>;
+
+  constructor(params: MultipleTargetsExtensionRunnerParams) {
+    this.extensionRunners = params.runners;
+  }
+
+  async run(): Promise<void> {
+    const promises = [];
+    for (const runner of this.extensionRunners) {
+      promises.push(runner.run());
+    }
+
+    await Promise.all(promises);
+  }
+
+  async reloadAllExtensions(): Promise<void> {
+    const promises = [];
+    for (const runner of this.extensionRunners) {
+      promises.push(runner.reloadAllExtensions());
+    }
+
+    await Promise.all(promises);
+  }
+
+  async reloadExtensionBySourceDir(sourceDir: string): Promise<void> {
+    const promises = [];
+    for (const runner of this.extensionRunners) {
+      promises.push(runner.reloadExtensionBySourceDir(sourceDir));
+    }
+
+    await Promise.all(promises);
+  }
+
+  registerCleanup(fn: Function): void {
+    const promises = [];
+    for (const runner of this.extensionRunners) {
+      promises.push(new Promise((resolve) => {
+        runner.registerCleanup(resolve);
```

Comment:
This is really confusing to me. Can't you just do `runner.registerCleanup(fn)` ? I'm also confused by `Promise.all(promises).then(fn, fn)` so maybe I'm missing it? Can you add a comment explaining what this is doing? :P